src/higley_lab_to_nwb/extractors/processed_imaging_extractor.py

```diff
@@ -0,0 +1,169 @@
+    """Load pixels matrix from MATLab file generated processing imaging data in 1p-imaging only configuration."""
+
+    with h5py.File(file_path, "r") as file:
+        pixels_matrix = file["dFoF"][process_type][:]
+
+        num_rows = int(file["R"][:])
+        num_columns = int(file["C"][:])
+
+    return np.nan_to_num(pixels_matrix), num_rows, num_columns
+
+
+def _get_mask_from_dual_imaging_configuration(file_path: str) -> np.ndarray:
+    with h5py.File(file_path, "r") as file:
+        mask = file["mask"][:]
+    return mask
+
+
+class ProcessedImagingExtractor(ImagingExtractor):
+    """Specialized extractor for reading processed imaging data from a MATLab file."""
+
+    extractor_name = "ProcessedImagingExtractor"
+    is_writable = True
+    mode = "file"
+
+    def __init__(
+        self,
+        file_path: PathType,
+        sampling_frequency: float,
+        process_type: Literal["dff_final", "dff_blue", "dff_uv", "blue", "uv", "green"],
+    ) -> None:
+        """Create a ProcessedImagingExtractor instance from a MATLab file.
+
+        Parameters
+        ----------
+        file_path : PathType
+            Path to the MATLab file.
+        sampling_frequency : float
+            The frequency at which the frames were sampled, in Hz.
+        process_type : str
+            The type of processed imaging data to extract.
+            Accepted values: "dff_final", "dff_blue", "dff_uv", "blue", "uv",  "green".
+        Notes
+        -----
+        """
+        super().__init__()
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        self._times = None
+
+        self._channel_names = process_type
+
+        if process_type in ["dff_final", "dff_blue", "dff_uv"]:
+            self._pixels_matrix, self._num_rows, self._num_columns = _load_data_from_dual_imaging_configuration(
+                file_path=file_path, process_type=process_type
+            )
+            mask = _get_mask_from_dual_imaging_configuration(file_path=file_path)
+        elif process_type in ["blue", "uv", "green"]:
+            self._pixels_matrix, self._num_rows, self._num_columns = _load_data_from_1p_imaging_configuration(
+                file_path=file_path, process_type=process_type
+            )
+
+        self._num_frames, number_of_pixels = self._pixels_matrix.shape
+
+        if number_of_pixels == self._num_rows * self._num_columns:
+            self._mask = None
+
+        elif number_of_pixels == np.count_nonzero(mask):
+            self._mask = mask
+
+        else:
+            raise ValueError(
+                f"Can't reconstruct frame from pixel matrix. The total number of pixels {number_of_pixels} does "
+                f"not match the frame dimension {self._num_rows} * {self._num_columns} or the non-zero elements in "
+                f"the frame mask {np.count_nonzero(mask)}."
+            )
```

Comment:
Should be checked if this returns the dtype of frame, but I think this is better than calling get_video() -> get_frames() -> _get_single_frame()
```suggestion
            )
            self._dtype = self._pixels_matrix[0].dtype
```